examples/elf/tests/Makefile

```diff
@@ -100,19 +100,10 @@ ifneq ($(CONFIG_EXAMPLES_ELF_NOSTRIP),y)
 endif
 endif
 
-# Create the file system image directory
-
-$(FSIMG_DIR):
-	$(Q) mkdir $(FSIMG_DIR)
-
-# Populate the file system image directory
-
-populate:  $(FSIMG_DIR) build install
```

Comment:
@masayuki2009 I think I found the root cause why object files get truncated:
The populate target above trigger build and install run parallelly, and then gcc may compile the same source file twice concurrently.
Please try my new PR, which modify three similar examples(elf/posix_spwan/sotest).

Reply:
@xiaoxiang781216 I tried the new PR, but it failed.

```
riscv64-unknown-elf-ld: task.o: file not recognized: file truncated                                                                                                                                    
Makefile:82: recipe for target 'task' failed                                                                                                                                                           
make[4]: *** [task] Error 1                                                                                                                                                                            
make[4]: Leaving directory '/mnt/m2ssd/opensource/github_xiang/apps/examples/elf/tests/task'                                                                                                           
Makefile:88: recipe for target 'task_install' failed                                                                                                                                                   
make[3]: *** [task_install] Error 2                                                                                                                                                                    
make[3]: Leaving directory '/mnt/m2ssd/opensource/github_xiang/apps/examples/elf/tests'                                                                                                                
Makefile:57: recipe for target 'tests/symtab.c' failed                                              
```